### PR TITLE
Alberto/atqr 92 refactor controller methods

### PR DESCRIPTION
**Pull request by albnunes**
I made endpoint responsible for a group of methods and the services in front are updated.

### CODE REVIEW

**Comment by albnunes** [comment]
https://linear.app/toki-labs/issue/ATQR-92/refactor-controller-methods

**Comment by yfernandes** [comment]
Hey @rairezende and @albnunes I'll be reviewing this one.  @albnunes since there are only two conflicting files I can take care of those for you

File: apps/atqr/api/src/app/dtos/index.ts [hunk 1]
```diff
@@ -2,3 +2,4 @@
 
 export * from './createChallenge.dto';
 export * from './updateCreditCardToken.dto';
+export * from './updateChallenge.dto'
```
Comment (yfernandes) [inline apps/atqr/api/src/app/dtos/index.ts]: A update challenge dto is imported but one is never created

File: apps/atqr/api/src/app/app.module.ts [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import { DeadlineMonitorService, NotificationService } from '@atqr/domain';
-import { Module } from '@nestjs/common';
+import { MiddlewareConsumer, Module, RequestMethod } from '@nestjs/common';
 import { ConfigModule, ConfigService } from '@nestjs/config';
 import { ChallengeController } from './challenge.controller';
 import { Mailer, PrismaService, StripeService } from './infra';
+import { PatchMiddleware } from './middleware/patchMiddleware';
```
Comment (yfernandes) [inline apps/atqr/api/src/app/app.module.ts]: There is no reference to a patch middleware, are you sure it really exists?

File: apps/atqr/api/src/app/app.module.ts [hunk 3]
```diff
@@ -29,6 +30,11 @@ import { ChallengeRepository, PlayerRepository } from './repositories';
...
       },
     },
+
   ],
 })
-export class AppModule {}
+export class AppModule {  configure(consumer: MiddlewareConsumer) {
+  consumer
+    .apply(PatchMiddleware)
+    .forRoutes({ path: 'challenge/:id', method: RequestMethod.PATCH });
```
Comment (yfernandes) [inline apps/atqr/api/src/app/app.module.ts]: Configuration for missing dependency.

**Review by yfernandes — CHANGES_REQUESTED** [review]
Major changes looks fine, but there are missing dependencies i can't find anywhere. Those have to be  resolved before being approved.

**Review by yfernandes — APPROVED** [review]
After changes everything is working!

**Pull request merged**